src/modules/order/order.service.ts

```diff
@@ -117,4 +152,13 @@ export class OrderService {
         );
         return orderList.toDtos();
     }
+
+    async getPromocode(promocodeName: string): Promise<IPromoCode> {
+        const promo: PromocodeNameDto = { name: promocodeName };
+        const isValid = await this.promocodeService.isPromoCodeValid(promo);
+        if (isValid) {
+            const promocode = this.promocodeService.getPromocodeByName(promo);
```

Comment:
If we combine these two methods on promocode microservice, we will get a GRPC error, because in case we have an invalid promo code, we cannot return null or undefined.